File: app/src/main/java/moe/shizuku/fontprovider/BaseActivity.java
package moe.shizuku.fontprovider;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentActivity;
import android.util.Log;
import android.view.MenuItem;

/**
 * Created by rikka on 2017/10/2.
 */

public abstract class BaseActivity extends FragmentActivity {

    private static boolean sFontInitialized = false;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        if (!sFontInitialized) {
            final long time = System.currentTimeMillis();

            FontProviderClient client = FontProviderClient.create(this);
            client.replace(FontRequests.DEFAULT_SANS_SERIF_FONTS, "Noto Sans CJK",
                    "sans-serif", "sans-serif-medium");

                    Log.d("Font", "replace costs " + (System.currentTimeMillis() - time) + "ms");

            sFontInitialized = true;
        }

        super.onCreate(savedInstanceState);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: sample/src/main/java/moe/haruue/test/fontprovider/BaseActivity.kt
package moe.haruue.test.fontprovider

import android.app.Activity
import android.os.Bundle
import moe.shizuku.fontprovider.FontProviderClient

abstract class BaseActivity: Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        if (!isFontInitialized) {
            FontProviderClient.create(this)?.let { client ->
                client.isNextRequestReplaceFallbackFonts = true
                client.replace("Noto Sans CJK", "sans-serif", "sans-serif-medium")
                client.replace("Noto Serif CJK", "serif", "serif-medium")
                isFontInitialized = true
            }
        }
        super.onCreate(savedInstanceState)
    }

    companion object {

        private var isFontInitialized = false

    }

}